### ISSUE TITLE
Treats Optional[object] as object

### ISSUE THREAD

**Issue by povilasb** [microsoft/pyright#2217]
**Describe the bug**

When a function returns `Optional[MyClass]` and then I use methods of `MyClass`, `pyright` does not warn me about not testing for `None`.

**To Reproduce**

Run mypy and pyright on this code snippet:

```
from typing import Optional


class Form:
    def set(self, key: str, val: str) -> "Form":
        return self


def make_form(text: str) -> Optional[Form]:
    if not text:
        return None
    return Form()


res = make_form("")
res.set("username", "user1")
```

**Expected behavior**

`mypy` says:
```
error: Item "None" of "Optional[Form]" has no attribute "set"
```
I would expect the same from `pyright`.

**command-line**

```bash
$ pyright --version
pyright 1.1.163
```

**Additional context**

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported. Pyright's "reportOptionalMemberAccess" diagnostic check reports the following error:

```
"set" is not a known member of "None" [reportOptionalMemberAccess]
```

Is it possible you have this diagnostic check disabled in your configuration?

**Comment by povilasb** [comment]
It was. Sorry for false positive :)
